docs/advanced-features/instrumentation.md

```diff
@@ -0,0 +1,38 @@
+---
+description: Learn how to instrument your Next.js app.
+---
+
+> **Note**: This feature is experimental. To use it, you must explicitly opt in by defining `experimental.instrumentationHook = true;` in your `next.config.js`.
+
+# `instrumentation.ts`
+
+If you export a function named `register` from this file, we will call that function whenever new server instance is bootstrapped.
+When your `register` function is deployed, it will be called continuously on each cold boot (but exactly once in each environment).
+
+Sometimes, it may be useful to import a file in your code because of the side effects it will cause. For example, you might import a file that defines a set of global variables, but never explicitly use the imported file in your code. You would still have access to the global variables the package has declared.
+
+The behavior is similar, but using imports with side-effects is hard to work with and has many downsides.
+
+**We recommend importing the `init` function from `package-init` instead of importing packages with side effects**. You can then call it from your custom `register` function defined in `instrumentation.ts`, as demonstrated in the example below:
+
+```ts
+// instrumentation.ts
+
+import { init } from 'package-init'
```

Comment:
What does `init` do?

Reply:
If the library needs initialization, it can be either through import side-effect or it exports a function that you can call. This should represent the second case.